src/pidROS.cpp

```diff
@@ -0,0 +1,299 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2008, Willow Garage, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/*
+  Author: Melonee Wise
+  Contributors: Dave Coleman, Jonathan Bohren, Bob Holmberg, Wim Meeussen
+  Desc: Implements a standard proportional-integral-derivative controller
+*/
+
+#include <algorithm>
+#include <cmath>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "control_toolbox/pidROS.hpp"
+
+namespace control_toolbox
+{
+
+PidROS::PidROS(rclcpp::Node::SharedPtr & node, std::string topic_prefix)
+: node_(node)
+{
+  if (topic_prefix.back() != '.' && !topic_prefix.empty()) {
+    topic_prefix_ = topic_prefix + ".";
+  } else {
+    topic_prefix_ = topic_prefix;
+  }
+
+  state_pub_ = node->create_publisher<control_msgs::msg::PidState>(
+    topic_prefix + "/pid_state", rclcpp::SensorDataQoS());
+  rt_state_pub_.reset(
+    new realtime_tools::RealtimePublisher<control_msgs::msg::PidState>(state_pub_));
+}
+
+PidROS::~PidROS()
```

Comment:
this is an empty desctructor, not even virtual. Can we not define it please? If you really wish to declare, can make it ` = default;` in the header but even that IMHO is too much for not adding anything but the default